### PR TITLE
Use redux components (removes p-flux)

### PR DESCRIPTION
**Pull request by crswty**
This PR finishes #155 . Bound action creators and dispatchers have been removed and all dispatch events have been connected into components.

I made the decision to pull all of the dispatch events in child components of show_retro up into show_retro for the time being as that is also where the state is connected. This means there is way too much going on in there but I think that's the best compromise for now.

In the future I would like to explore making connected child components. I think this also mean thinking about adding some integration level tests that test the components interact correctly with redux.

* [X] I have reviewed the [contributing guide](https://github.com/pivotal/postfacto/blob/master/CONTRIBUTING.md)

* [X] I have made this pull request to the `master` branch

* [X] I have run all the tests using `./test.sh`.

* [X] I have added the [copyright headers](https://github.com/pivotal/postfacto/blob/master/license-header.txt) to each new file added

* [X] I have given myself credit in the [humans.txt](https://github.com/pivotal/postfacto/blob/master/humans.txt) file (assuming I want to)

### CODE REVIEW

File: web/package.json [hunk 1]
```diff
@@ -9,7 +9,6 @@
     "material-ui": "^0.20.2",
     "mixpanel-browser": "^2.26.0",
     "node-sass": "^4.12.0",
-    "p-flux": "^1.1.1",
```
Comment (crswty) [inline web/package.json]: 🎉

File: web/src/components/retro-show/retro_cable.test.jsx [hunk 2]
```diff
@@ -32,17 +32,18 @@
...
 
 describe('RetroCable', () => {
   let retroCableDOM;
+  let websocketRetroDataReceived;
 
   beforeEach(() => {
+    websocketRetroDataReceived = jest.fn();
     const cable = ActionCable.createConsumer('wss://websocket/url');
-    retroCableDOM = mount(<RetroCable cable={cable} retro_id="retro-slug-123"/>);
+    retroCableDOM = mount(<RetroCable cable={cable} retro_id="retro-slug-123" websocketRetroDataReceived={websocketRetroDataReceived}/>);
```
Comment (textbook) [inline web/src/components/retro-show/retro_cable.test.jsx]: Is there any way to test whether `websocketRetroDataReceived` gets invoked at the right time?

File: web/package.json (hunk 1 above)
Comment (textbook) [reply web/package.json]: 🔥

File: web/src/components/retro-show/retro_cable.test.jsx (hunk 2 above)
Comment (crswty) [reply web/src/components/retro-show/retro_cable.test.jsx]: Yea, think I removed the 'test' for this as I didn't think it really tested that. I would probably revisit this if I get round to writing some component integration tests.

**Pull request merged**